src/export/InstanceExporter.ts

```diff
@@ -836,7 +852,31 @@ export class InstanceExporter implements Fishable {
     this.pkg.instances.push(instanceDef);
 
     // Once all rules are set, we should ensure that we did not add a duplicate profile URL anywhere
-    if (instanceDef.meta?.profile) instanceDef.meta.profile = uniq(instanceDef.meta.profile);
+    // if any of the duplicates have child elements, such as extension, merge them
+    if (instanceDef.meta?.profile) {
+      if (instanceDef.meta._profile) {
+        const complexProfiles = zip(instanceDef.meta.profile, instanceDef.meta._profile);
+        const uniqueProfiles = uniqWith(complexProfiles, (a, b) => {
+          return a[0] != null && a[0] === b[0];
+        });
+        const removedProfiles = differenceWith(complexProfiles, uniqueProfiles, isEqual);
+        removedProfiles.forEach(rp => {
+          const homeIndex = uniqueProfiles.findIndex(up => up[0] === rp[0]);
+          mergeWith(uniqueProfiles[homeIndex][1], rp[1], (a, b) => {
+            // arrays are concatenated, primitives prefer the existing value if both values are non-null
+            // this makes use of the fact that typeof null === 'object'
+            if (isArray(a) && isArray(b)) {
+              return a.concat(b);
+            } else if (typeof a !== 'object' || typeof b !== 'object') {
+              return a ?? b;
+            }
+          });
+        });
+        [instanceDef.meta.profile, instanceDef.meta._profile] = unzip(uniqueProfiles);
```

Comment:
Wow. What a block of code. Things like this remind me that I should try to keep myself more familiar w/ lodash.